python/triton/compiler/compiler.py

```diff
@@ -85,12 +85,17 @@ def optimize_ttgir(mod, num_stages, arch):
         pm.add_tritongpu_accelerate_matmul_pass(80)
     pm.add_tritongpu_remove_layout_conversions_pass()
     pm.add_tritongpu_optimize_dot_operands_pass()
-    pm.add_tritongpu_pipeline_pass(num_stages)
+    if num_stages == 0 and is_hip() and gpu_has_mfma():
+        pm.add_tritongpu_stream_pipeline_pass()
+        pm.add_canonicalizer_pass()
+    else:
+        pm.add_tritongpu_pipeline_pass(num_stages)
     pm.add_tritongpu_prefetch_pass()
     pm.add_tritongpu_optimize_dot_operands_pass()
     pm.add_tritongpu_remove_layout_conversions_pass()
     pm.add_tritongpu_decompose_conversions_pass()
-    pm.add_tritongpu_reorder_instructions_pass()
+    if num_stages != 0:
```

Comment:
This pass moves the convert_layouts too early in the loop, potentially causing bad logic.

Reply:
Yes, this pass does produce wrong values because it moves the convert_layouts over each other. This should be fixed in the pass, I am now working on fixing this bug.

Reply:
We do reply on this `reorderInstruction` pass to keep q tensor in registers for FA fwd, which proves to be an important optimization for head dim=64.

Reply:
Yes, stream-pipelining should not be enabled for FA yet.